metals/src/main/scala/scala/meta/internal/metals/testProvider/frameworks/SpecsTestFinder.scala

```diff
@@ -0,0 +1,27 @@
+package scala.meta.internal.metals.testProvider.frameworks
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.testProvider.TestCaseEntry
+import scala.meta.internal.mtags
+import scala.meta.internal.parsing.Trees
+import scala.meta.internal.semanticdb.SymbolInformation
+import scala.meta.internal.semanticdb.TextDocument
+import scala.meta.internal.semanticdb.TypeRef
+import scala.meta.io.AbsolutePath
+import scala.meta.transversers._
+import scala.reflect.NameTransformer
+
+object SpecsTestFinder {
+
+  def collectTestCases(path: AbsolutePath, trees: Trees) = {
+    val treeOpt = trees.get(path)
+    treeOpt.map { tree =>
+      tree.collect {
+        case in @ meta.Term.ApplyInfix(lhs, meta.Name("in"), _, _) => {
+          println(s"RFERGUSON found an `in` at ${in.pos.startLine}")
```

Comment:
There's actually more here on my current branch, but this was how it started. I left it like this for this branch because I'm thinking about making these changes more accessible to everyone and don't want the buttons to appear and do nothing.